pkg/cloud/vsphere/vsphere.go

```diff
@@ -89,3 +95,204 @@ func NewProviderAssets(config config.OperatorConfig) (common.CloudProviderAssets
+		return "", fmt.Errorf("failed to modify the provided configuration: %w", err)
+	}
+
+	return buf.String(), nil
+}
+
+func setLabelsSection(cfg *ini.File) error {
+	labels := map[string]string{"region": "openshift-region", "zone": "openshift-zone"}
+
+	labelsSection, err := setOrGetSection(cfg, "Labels")
+	if err != nil {
+		return err
+	}
+
+	for k, v := range labels {
+		_, err = setOrGetKeyValue(labelsSection, k, v, true)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Check to make sure the [Labels] section doesn't have
+	// any incorrect keys
+	for _, k := range labelsSection.Keys() {
+		if _, ok := labels[k.Name()]; !ok {
+			klog.Warningf("Key %s should not be in [%s] section, removing.", k.Name(), labelsSection.Name())
+			labelsSection.DeleteKey(k.Name())
+		}
+	}
+
+	return nil
+}
+
+func setVCenterPortKey(vCenterSection *ini.Section, port int32) error {
+	_, err := setOrGetKeyValue(vCenterSection, "port", strconv.FormatInt(int64(port), 10), true)
+	return err
+}
+
+func setVCenterDatacentersKey(vCenterSection *ini.Section, datacenter string) error {
+	// Get existing key (don't overwrite) or get new key
+	key, err := setOrGetKeyValue(vCenterSection, "datacenters", datacenter, false)
+	if err != nil {
+		return err
+	}
+
+	datacenters := key.String()
+
+	if !strings.Contains(datacenters, datacenter) {
+		klog.Infof("Appending %s to existing datacenters key value %s", datacenter, datacenters)
+		datacenters = fmt.Sprintf("%s,%s", datacenters, datacenter)
+		key.SetValue(datacenters)
+	}
+
+	return nil
+}
+
+func setNodesSection(cfg *ini.File, nodeNetworking *configv1.VSpherePlatformNodeNetworking) error {
+	keyValues := map[string]string{
+		"external-vm-network-name":             nodeNetworking.External.Network,
+		"internal-vm-network-name":             nodeNetworking.Internal.Network,
+		"external-network-subnet-cidr":         strings.Join(nodeNetworking.External.NetworkSubnetCIDR[:], ","),
+		"exclude-external-network-subnet-cidr": strings.Join(nodeNetworking.External.ExcludeNetworkSubnetCIDR[:], ","),
+		"internal-network-subnet-cidr":         strings.Join(nodeNetworking.Internal.NetworkSubnetCIDR[:], ","),
+		"exclude-internal-network-subnet-cidr": strings.Join(nodeNetworking.Internal.ExcludeNetworkSubnetCIDR[:], ","),
+	}
+
+	nodesSection, err := setOrGetSection(cfg, "Nodes")
+	if err != nil {
+		return err
+	}
+
+	for k, v := range keyValues {
+		if v != "" {
+			_, err = setOrGetKeyValue(nodesSection, k, v, true)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	// if there are no new or existing keys on [Nodes] section
+	// [Nodes] should not exist, delete section
+	if len(nodesSection.Keys()) == 0 {
+		cfg.DeleteSection("Nodes")
+	}
+
+	return nil
+}
+
+func setOrGetSection(cfg *ini.File, name string) (*ini.Section, error) {
+	if cfg.HasSection(name) {
+		klog.Infof("%s section found...", name)
+		return cfg.GetSection(name)
+	} else {
+		klog.Infof("Creating new section: %s", name)
+		return cfg.NewSection(name)
```

Comment:
Logs here are probably too chatty at `V(0)`, this is pretty debug level IMO

Reply:
fixed